cloudflare/resource_cloudflare_teams_route.go

```diff
@@ -0,0 +1,144 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareTeamsRoute() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tunnel_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"comment": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+		Create: resourceCloudflareTeamsRouteCreate,
+		Read:   resourceCloudflareTeamsRouteRead,
+		Update: resourceCloudflareTeamsRouteUpdate,
+		Delete: resourceCloudflareTeamsRouteDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareTeamsRouteImport,
+		},
+	}
+}
+
+func resourceCloudflareTeamsRouteRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	tunnelRoute, err := client.GetTunnelRouteForIP(context.Background(), cloudflare.TunnelRoutesForIPParams{
+		AccountID: d.Get("account_id").(string),
+		Network:   d.Get("network").(string),
+	})
+	if err != nil {
+		// TODO: implement logic for missing route
+
+		return fmt.Errorf("error reading Tunnel Route for Network %q: %w", d.Id(), err)
+	}
+
+	d.Set("tunnel_id", tunnelRoute.TunnelID)
+	d.Set("network", tunnelRoute.Network)
+
+	if len(tunnelRoute.Comment) > 0 {
+		d.Set("comment", tunnelRoute.Comment)
+	}
+
+	return nil
+}
+
+func resourceCloudflareTeamsRouteCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	resource := cloudflare.TunnelRoutesCreateParams{
+		AccountID: d.Get("account_id").(string),
+		TunnelID:  d.Get("tunnel_id").(string),
```

Comment:
given the `TunnelID` is read only, this is probably better suited to be used as the `SetID` call.

```suggestion
```